File: tests/rosetta/transpiler/Kotlin/add-a-variable-to-a-class-instance-at-runtime.kt
fun input(): String = readLine() ?: ""

data class SomeStruct(val runtimeFields: MutableMap<String, String>)
fun user_main(): Unit {
    var ss: SomeStruct = SomeStruct(runtimeFields = mutableMapOf<Any, Any>())
    println("Create two fields at runtime: \n")
    var i: Int = 1
    while (i <= 2) {
        println(("  Field #" + i.toString()) + ":\n")
        println("       Enter name  : ")
        val name: String = input()
        println("       Enter value : ")
        val value: String = input()
        var fields = ss.runtimeFields
        (fields as MutableMap<String, Any?>)[name] = value
        ss.runtimeFields = fields
        println("\n")
        i = i + 1
    }
    while (true) {
        println("Which field do you want to inspect ? ")
        val name: String = input()
        if (name in ss.runtimeFields) {
            val value: String = (ss.runtimeFields)[name]!!
            println(("Its value is '" + value) + "'")
            return
        } else {
            println("There is no field of that name, try again\n")
        }
    }
}

fun main() {
    user_main()
}

File: tests/rosetta/transpiler/php/add-a-variable-to-a-class-instance-at-runtime.php
<?php
function main() {
  $ss = ["runtimeFields" => []];
  echo "Create two fields at runtime: \n", PHP_EOL;
  $i = 1;
  while ($i <= 2) {
  echo "  Field #" . json_encode($i, 1344) . ":\n", PHP_EOL;
  echo "       Enter name  : ", PHP_EOL;
  $name = trim(fgets(STDIN));
  echo "       Enter value : ", PHP_EOL;
  $value = trim(fgets(STDIN));
  $fields = $ss["runtimeFields"];
  $fields[$name] = $value;
  $ss["runtimeFields"] = $fields;
  echo "\n", PHP_EOL;
  $i = $i + 1;
};
  while (true) {
  echo "Which field do you want to inspect ? ", PHP_EOL;
  $name = trim(fgets(STDIN));
  if (array_key_exists($name, $ss["runtimeFields"])) {
  $value = $ss["runtimeFields"][$name];
  echo "Its value is '" . $value . "'", PHP_EOL;
  return;
} else {
  echo "There is no field of that name, try again\n", PHP_EOL;
}
};
}
main();
